c10/core/TensorImpl.h

```diff
@@ -138,6 +138,44 @@ struct C10_API AutogradMetaInterface {
   virtual ~AutogradMetaInterface();
 };
 
+// NOTE [ Version Counter Sharing ]
+//
+// Every Tensor has a version counter. Version counters are incremented
+// whenever the data or shape of a tensor changes through Variable operations.
+// These are typically in-place operations. Version counters are used to
+// detect modifications to saved variables which would result in incorrect
+// gradient calculations. Version counters may be shared between Variables:
+//
+// 1. A view shares the version counter of the base Variable,
+// 2. Detached variables share the version counter of the source,
+// 3. Unpacked saved variables share the version counter of the source.
+//
+// Version counters are not shared when:
+// 1. We replace a `Variable`'s underlying `Tensor` by calling `set_data(...)`.
+//
+// Note that we explicitly don't increment the version counter in non-Variable
+// operations, because it is a useful escape hatch when we want to change the
+// content of a Variable without invalidating it in the autograd graph.
```

Comment:
This doesn't describe *why* TensorImpl isn't on AutogradMeta.

Reply:
Fixed